### DISCUSSION TITLE
Is there a guarantee that to get one and only one `QUIC_STREAM_EVENT_SEND_COMPLETE` event for each sucessful `StreamSend()` call?

### DISCUSSION THREAD

**Discussion by Wizmann** [microsoft/msquic#2209] [Q&A] [answered]
Scenario 1:

Say we've called `StreamSend()` with a 100 bytes buffer, is there a guarantee that we will get one and only one `QUIC_STREAM_EVENT_SEND_COMPLETE` for successfully sending that buffer? 

Scenario 2:

We called `StreamSend()` twice with 100 bytes buffer and 200 bytes buffer, is there any possiblity that we'll get a `QUIC_STREAM_EVENT_SEND_COMPLETE` event with a "successfully sent 300 byte. Or we will get two events for sending each one of the buffers.

Also, does `QUIC_PARAM_CONN_SEND_BUFFERING` will have any impact on this behavior?

p.s. It could be better if you can add some explicit instruction about this in `Stream.md`.

**Comment by nibanks** [comment, marked as answer]
Every successful call to `StreamSend` is **guaranteed** to get a matching, single `QUIC_STREAM_EVENT_SEND_COMPLETE` event; no matter what. Send buffering doesn't change this guarantee, only when the event is delivered.

I agree that clarifying this point in both https://github.com/microsoft/msquic/blob/main/docs/Streams.md and https://github.com/microsoft/msquic/blob/main/docs/api/StreamSend.md would probably be a good idea.
